### ISSUE TITLE
Specify Custom/Mandatory fields of .eb file in the configuration file

### ISSUE THREAD

**Issue by sgabenov** [easybuilders/easybuild-framework#3068]
In our project, we need to have some additional fields in easybuild files to be set and be mandatory, as our further workflow depends on these fields, like: 'application owner', 'sync sites', 'add to catalog' and others. This is done by patching EasyBuild framework, but I think this is quite a usual thing, so it would be great if we can specify such additional mandatory fields in easybuild config file.

**Comment by ocaisa** [comment]
Is it that you simply want to see these values appear in the final module file? I think the right way to approach this would be for you to implement the appropriate hooks to handle this, see https://easybuild.readthedocs.io/en/latest/Hooks.html

That way you can even handle special cases and take actions based on the values provided there through other hooks. For the simple case where you just want to check values and add them to the module file you would just need the `parse_hook` and the `pre_module_hook` (I think). Since you already patch the framework to handle this, then it should be just translating the patch into hooks.

**Comment by boegel** [comment]
I agree that doing this through hooks is probably the best way forward for this.

@sgabenov Do let us know if you have any questions on this, or if you need any help.
